frontend/app_flowy/packages/appflowy_editor/lib/src/extensions/text_node_extensions.dart

```diff
@@ -29,40 +29,42 @@ extension TextNodeExtension on TextNode {
   }
 
   bool allSatisfyLinkInSelection(Selection selection) =>
-      allSatisfyInSelection(selection, BuiltInAttributeKey.href, (value) {
+      allSatisfyInSelection(selection, BuiltInAttributeKey.href, <bool>(value) {
         return value != null;
       });
 
   bool allSatisfyBoldInSelection(Selection selection) =>
-      allSatisfyInSelection(selection, BuiltInAttributeKey.bold, (value) {
+      allSatisfyInSelection(selection, BuiltInAttributeKey.bold, <bool>(value) {
         return value == true;
       });
 
   bool allSatisfyItalicInSelection(Selection selection) =>
-      allSatisfyInSelection(selection, BuiltInAttributeKey.italic, (value) {
+      allSatisfyInSelection(selection, BuiltInAttributeKey.italic,
+          <bool>(value) {
         return value == true;
       });
 
   bool allSatisfyUnderlineInSelection(Selection selection) =>
-      allSatisfyInSelection(selection, BuiltInAttributeKey.underline, (value) {
+      allSatisfyInSelection(selection, BuiltInAttributeKey.underline,
+          <bool>(value) {
         return value == true;
       });
 
   bool allSatisfyStrikethroughInSelection(Selection selection) =>
       allSatisfyInSelection(selection, BuiltInAttributeKey.strikethrough,
-          (value) {
+          <bool>(value) {
         return value == true;
       });
 
   bool allSatisfyCodeInSelection(Selection selection) =>
-      allSatisfyInSelection(selection, BuiltInAttributeKey.code, (value) {
+      allSatisfyInSelection(selection, BuiltInAttributeKey.code, <bool>(value) {
         return value == true;
       });
 
   bool allSatisfyInSelection(
     Selection selection,
     String styleKey,
-    bool Function(dynamic value) test,
+    bool Function<T>(T value) test,
```

Comment:
Why do you need to use generics `T` to constrain the parameters? Is there any benefit?